lib/f_core/net/network/c_ipv4.cpp

```diff
@@ -0,0 +1,50 @@
+// Self Include
+#include "f_core/net/network/c_ipv4.h"
+
+// Zephyr Includes
+#include <zephyr/logging/log.h>
+#include <zephyr/net/net_if.h>
+
+LOG_MODULE_REGISTER(CIPv4);
+
+CIPv4::CIPv4(const char* ip) : ip(ip), netIface(*net_if_get_default()) {
+    initialize();
+};
+
+CIPv4::CIPv4(const char* ip, net_if *net_iface) : ip(ip), netIface(*net_iface) {
+    initialize();
+};
+
+CIPv4::CIPv4(const char* ip, const device* dev) : ip(ip), netIface(*net_if_lookup_by_dev(dev)) {
```

Comment:
Same as above, could delegate to `CIPv4::CIPv4(const char* ip, net_if *net_iface)`